### ISSUE TITLE
Backward compatibility for export / map containerization

### ISSUE THREAD

**Issue by AlexDaniel** [rakudo/rakudo#3133]
* problem-solving ticket: https://github.com/perl6/problem-solving/issues/74
* Rakudo PR in response to that: https://github.com/rakudo/rakudo/pull/3081
* This is the commit from the PR that broke some things in the ecosystem: https://github.com/rakudo/rakudo/pull/3081/commits/2ee82afe6e6713c26b929e2855ea05d8eb3b9fee
* Modules that were affected: ASN::META, OEIS (discovered in https://github.com/rakudo/rakudo/issues/3113)
* PRs that fixed the modules: https://github.com/Altai-man/ASN-META/pull/1 https://github.com/taboege/p6-OEIS/pull/1

So the two modules that were affected are already fixed by @vrurg++, but it makes me wonder if we're trying to slip a breaking change into the next release… Especially given that we have evidence that there is code out there that relied on the previous behavior.

**Comment by vrurg** [comment]
I was always saying that what's been done is a fix, not a breakage:

```
⇒ perl6 -e 'my $b=42; my @a is List = [$b]; note @a[0].VAR.^name'
Scalar
⇒ perl6 -e 'my $b=42; my %a is Map = a=>$b; note %a<a>.VAR.^name'
Int
⇒ perl6 -v
This is Rakudo version 2019.03.1 built on MoarVM version 2019.03
implementing Perl 6.d.
```

Implicit decontainerization by `Map` was a totally incorrect thing.

**Comment by lucasbuchala** [comment]
Related: https://github.com/rakudo/rakudo/issues/3168, https://github.com/perl6/problem-solving/issues/99

**Comment by vrurg** [comment]
rakudo/rakudo#3174 is probably resolves @AlexDaniel  concerns.

**Comment by vrurg** [comment]
This one can be closed with merge of #3174. Besides, there is also [my comment](https://github.com/perl6/problem-solving/issues/99#issuecomment-532467970) in perl6/problem-solving#99 which is addressing this issue too.

**Comment by AlexDaniel** [comment]
This seems to be alright now. Thank you, @vrurg.
